modules/phase_field/src/postprocessors/GrainTracker.C

```diff
@@ -417,13 +432,17 @@ GrainTracker::trackGrains()
    * Next we need to look at our new list and see which grains weren't matched up.  These are new grains.
    */
   for (auto map_num = decltype(_maps_size)(0); map_num < _maps_size; ++map_num)
-    for (unsigned int i = 0; i < _feature_sets[map_num].size(); ++i)
-      if (_feature_sets[map_num][i] && _feature_sets[map_num][i]->_status == NOT_MARKED)
+    for (auto feature_num = decltype(_feature_sets[map_num].size())(0);
+         feature_num < _feature_sets[map_num].size(); ++feature_num)
+      // If it's not in the index list, it hasn't been transferred
+      if (new_grain_idx_to_existing_grain_idx.find(std::make_pair(map_num, feature_num)) == new_grain_idx_to_existing_grain_idx.end())
       {
+        mooseAssert(_feature_sets[map_num][i]._status == NOT_MARKED, "Feature in wrong state, logic error");
```

Comment:
Damn. I really need to switch to "devel" mode
On Sat, Jun 11, 2016 at 11:40 AM Daniel Schwen <notifications@github.com>
wrote:

> In modules/phase_field/src/postprocessors/GrainTracker.C
> <https://github.com/idaholab/moose/pull/7161#discussion_r66709167>:
>
> >        {
> > +        mooseAssert(_feature_sets[map_num][i]._status == NOT_MARKED, "Feature in wrong state, logic error");
>
> This line causes the compile error in debug. i is the index of the old
> for loop you removed. Needs to be feature_num.
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/idaholab/moose/pull/7161/files/f76d458bebb39499d88c62360a08500cbff860ef#r66709167>,
> or mute the thread
> <https://github.com/notifications/unsubscribe/AC5XIPvMLaNOyEbwyP5LfLMi9ugUEc2Fks5qKvMqgaJpZM4IwRk6>
> .
>